### PR TITLE
Refactor of Server 

### PR DESCRIPTION
**Pull request by deduced**
Changes made: 
* Revamped the backend models
* Updated asyncGetUserInfo to create the user after getting info from Auth0 if the user is not found in the DB. 
* Refactor /transactions endpoint to create items, accounts, and transactions for a given userId. This means that to call the get on /transactions, you must pass a userId. 
* General cleanup

What is not done: 
* Need to figure out how to chain the creation of transactions for a new user properly. Too late to continue now.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -233,181 +249,133 @@ app.use(
...
+//       }
+//     });
+//   });
+// });
 
 // Retrieve Transactions for an Item
 // https://plaid.com/docs/#transactions
 // NOTE: modified and used to seed database with plaid sandbox data
-app.get("/transactions", function(request, response, next) {
+app.get("/transactions", function(request, response) {
```
Comment (jpdaprato) [inline server/index.js]: Let's consider rolling this process up into a function and calling it in the asynCreateItem function at the createItem GraphQL endpoint.

File: server/index.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ const AUTH0_API_TOKEN = process.env.AUTH0_API_TOKEN;
 // PLAID API
 // We store the access_token in memory - in production, store it in a secure
 // persistent data store
-let ACCESS_TOKEN = null;
+let ACCESS_TOKEN = process.env.ACCESS_TOKEN;
```
Comment (jpdaprato) [inline server/index.js]: ACCESS_TOKEN and ITEM_ID will be unique to each item and are returned by the Plaid API at the GraphQL createItem endpoint after exchanging the publicToken received from the React client, and thus should not be stored as environmental variables.

File: src/components/LinkItems.jsx [hunk 3]
```diff
@@ -25,9 +28,9 @@ class LinkItems extends React.Component {
     return (
       <PlaidLink
         clientName="cygnustechnologies"
-        env="development"
+        env={process.env.PLAID_ENV}
         product={["auth", "transactions"]}
-        publicKey="88a038c0956987b0027438f7596d9e"
+        publicKey={process.env.PLAID_PUBLIC_KEY}
```
Comment (jpdaprato) [inline src/components/LinkItems.jsx]: Nice! Wasn't sure how to do this using dotenv from React.

File: server/index.js (hunk 2 above)
Comment (deduced) [reply server/index.js]: This was just for testing with sandbox mode as it requires a non-null value (any value will do). I already set it back locally.

File: server/index.js (hunk 1 above)
Comment (deduced) [reply server/index.js]: I am revamping this structure a bit so will ignore this. When I push up you will see a model/controller/routes approach and we can then refactor appropriately. For now using the same rest endpoint but separating it out in a much better way. We can then roll into graphql resolvers.

File: server/controllers/pocketgoblin.controller.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+const result = dotenv.config();
+if (result.error) {
+  throw result.error;
+}
+
+let ACCESS_TOKEN = process.env.ACCESS_TOKEN;
+
+const controller = {
+  getTransactions: (request, response) => {
```
Comment (jpdaprato) [inline server/controllers/pocketgoblin.controller.js]: Rename to: saveItemData; getting transactions is merely a means to an end here. (Data needed by the frontend is separated out into separate endpoints in graphql server.) 
TODO: pull out each create/save function (called sequentially in the promise chain) for each table as helper functions.

File: server/routes/pocketgoblin.routes.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+const ctrl = require("../controllers/pocketgoblin.controller.js");
+module.exports = function(app) {
+  app.get("/transactions", ctrl.getTransactions);
```
Comment (jpdaprato) [inline server/routes/pocketgoblin.routes.js]: Replace w/ graphQL endpoint?

File: server/plaid/index.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+const plaid = require("plaid");
+const dotenv = require("dotenv");
+
+const result = dotenv.config();
+if (result.error) {
+  throw result.error;
+}
+const PLAID_CLIENT_ID = process.env.PLAID_CLIENT_ID;
+const PLAID_SECRET = process.env.PLAID_SECRET;
```
[the remark this answers is not in the document]
Comment (jpdaprato) [reply server/plaid/index.js]: Changed my mind on this one. Not as easy as I had thought and not worth the trouble.

**Pull request merged**